tests/components/recorder/test_util.py

```diff
@@ -1117,3 +1128,106 @@ async def test_resolve_period(hass: HomeAssistant) -> None:
             }
         }
     ) == (now - timedelta(hours=1, minutes=25), now - timedelta(minutes=25))
+
+
+NonRetryable = OperationalError(None, None, None, "")
+Retryable = OperationalError(None, None, BaseException(RETRYABLE_MYSQL_ERRORS[0], ""))
+
+
+@pytest.mark.parametrize(
+    ("side_effect", "expected_result", "nbr_calls", "dialect"),
+    [
+        (None, does_not_raise(), 1, SupportedDialect.MYSQL),
+        (ValueError, pytest.raises(ValueError), 1, SupportedDialect.MYSQL),
+        (NonRetryable, pytest.raises(OperationalError), 1, SupportedDialect.MYSQL),
+        (Retryable, pytest.raises(OperationalError), 5, SupportedDialect.MYSQL),
+        (NonRetryable, pytest.raises(OperationalError), 1, SupportedDialect.SQLITE),
+        (Retryable, pytest.raises(OperationalError), 1, SupportedDialect.SQLITE),
+    ],
+)
+def test_database_job_retry_wrapper(
+    side_effect: Any, expected_result: Callable, nbr_calls: int, dialect: str
```

Comment:
`expected_result` is a context manager, not a Callable:

```suggestion
    side_effect: Any, expected_result: AbstractContextManager, nbr_calls: int, dialect: str
```